community/cypher/frontend-3.0/src/main/scala/org/neo4j/cypher/internal/frontend/v3_0/ast/Function.scala

```diff
@@ -182,4 +182,26 @@ abstract class AggregatingFunction extends Function {
     when(ctx == ast.Expression.SemanticContext.Simple) {
       SemanticError(s"Invalid use of aggregating function $name(...) in this context", invocation.position)
     } chain invocation.arguments.semanticCheck(ctx) chain semanticCheck(ctx, invocation)
+
+
+  /*
+   * Checks so that the expression is in the range [min, max]
+   */
+  protected def checkRange(expression: Expression, min: Double, max: Double): SemanticCheck = {
+    expression match {
+      case d: DoubleLiteral if d.value >= min || d.value <= max =>
+        SemanticCheckResult.success
+      case d: DoubleLiteral =>
+        SemanticError(s"Invalid input '${d.value}' is not a valid argument, must be a number in the range 0.0 to 1.0",
+          d.position)
+      case l: Literal =>
+        SemanticError(s"Invalid input '${
+          l.asCanonicalStringVal
+        }' is not a valid argument, must be a number in the range 0.0 to 1.0", l.position)
+
+      //for other types we'll have to wait until runtime to fail
+      case _ => SemanticCheckResult.success
```

Comment:
I renamed the method, it sounded more general than what it is. It checks so that the number is in the range `[0.0, 1.0]` which not many integers are but you caught the bug that we should probably allow `percentileDisc(n.prop, 0)` and `percentileProp(n.prop, 1)`, :+1: